source/extensions/common/wasm/stats_handler.h

```diff
@@ -110,7 +110,8 @@ class LifecycleStatsHandler {
   COUNTER(vm_reload)                                                                               \
   COUNTER(vm_reload_backoff)                                                                       \
   COUNTER(vm_reload_success)                                                                       \
-  COUNTER(vm_reload_failure)
+  COUNTER(vm_reload_failure)                                                                       \
+  COUNTER(vm_failed)
```

Comment:
curl -s "http://localhost:9901/stats?filter=wasm"
wasm.envoy.wasm.runtime.v8.active: 18
wasm.envoy.wasm.runtime.v8.created: 19
wasm.remote_load_cache_entries: 0
wasm.remote_load_cache_hits: 0
wasm.remote_load_cache_misses: 0
wasm.remote_load_cache_negative_hits: 0
wasm.remote_load_fetch_failures: 0
wasm.remote_load_fetch_successes: 0
**wasm.vm_failed: 1**
wasm.vm_reload: 0
wasm.vm_reload_backoff: 0
wasm.vm_reload_failure: 0
wasm.vm_reload_success: 0

Reply:
Why is the plugin_name not rendering in the stats? I checked the code and it should be putting plugin name there.

Reply:
That's because I wasn't naming the extension in my local test. After adding a name, I can see it.
```
http_filters:
- name: envoy.filters.http.wasm
  typed_config:
    "@type": type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
    config:
      name: test_vm
      vm_config:
        runtime: "envoy.wasm.runtime.v8"
        code:
          local:
            filename: "/etc/envoy/extension.wasm"
```

curl -s "http://localhost:9901/stats?filter=wasm"
wasm.envoy.wasm.runtime.v8.active: 3
wasm.envoy.wasm.runtime.v8.created: 5
wasm.**test_vm**.vm_failed: 1
wasm.test_vm.vm_reload: 0
wasm.test_vm.vm_reload_backoff: 0
wasm.test_vm.vm_reload_failure: 0
wasm.test_vm.vm_reload_success: 1